bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogicFindAllStatus.java

```diff
@@ -0,0 +1,25 @@
+package org.eclipse.ui.internal.findandreplace;
+
+/**
+ * @since 1.0
+ */
+public class FindReplaceLogicFindAllStatus implements IFindReplaceLogicStatus {
+	private int selectCount;
+
+	public FindReplaceLogicFindAllStatus(int selectCount) {
+		if (selectCount <= 0) {
+			// invalid value - what to do? Throw an exception?! @HeikoKlare
```

Comment:
@HeikoKlare

Reply:
No need to check that. Even though it does not make sense semantically, the code will even work with a negative value. In case `getSelectCount()` would specify that the returned value is positive, the constructor would need to document an expected positive value as well.

In general, it  is not mandatory to validate all input parameters but depends on the situation whether that makes sense. E.g., a public API may validate input to ensure that nothing fails internally when operating on erroneous inputs, while some internal method (like this) may perform no unnecessary validation if the callers at hand are ensured to fulfill the method's contract anyway.